temporal-sdk/src/main/java/io/temporal/client/UpdateHandle.java

```diff
@@ -51,4 +52,14 @@
    * @return future completed with the result of the update or an exception
    */
   CompletableFuture<T> getResultAsync();
+
+  /**
+   * Returns a {@link CompletableFuture} with the update workflow execution request result
+   * potentially waiting for the update to complete.
+   *
+   * @param timeout maximum time to wait and perform the background long polling
+   * @param unit unit of timeout
+   * @return future completed with the result of the update or an exception
+   */
+  CompletableFuture<T> getResultAsync(long timeout, TimeUnit unit);
```

Comment:
Re https://github.com/temporalio/sdk-go/pull/1088#discussion_r1166787604, we also need this in Java at some point. We need a way to wait on a workflow handle to reach a stage even after submitted to server. But that doesn't have to be this PR.

Reply:
Yes when the server supports that we can have another method that waits on accepted.